src/ICeWall/src/ICeWall.tcl

```diff
@@ -3561,26 +3904,1336 @@ namespace eval ICeWall {
+      dict set padcell cell [verify_placement [dict get $padcell cell] $cell_name cell]
+    }
+
+    # Verify bondpad
+    if {[dict exists $padcell bondpad]} {
+      set bondpad_cell_ref [dict get $library types bondpad]
+      if {![dict exists $library cells $bondpad_cell_ref]} {
+        utl::error PAD 133 "Bondpad cell $bondpad_cell_ref not found in library definition"
+      }
+      if {[dict exists $padcell bondpad orient]} {
+        if {[dict exists $library cells $bondpad_cell_ref orient]} {
+          if {[dict exists $library cells $bondpad_cell_ref orient $side_name]} {
+            set expected_orient [dict get $library cells $bondpad_cell_ref orient $side_name]
+          } else {
+            if {[llength [dict get $library cells $bondpad_cell_ref orient]] == 1} {
+              set expected_orient [dict get $library cells $bondpad_cell_ref orient]  
+            } else {
+              utl::error PAD 134 "Unexpected value for orient attribute in library definition for $bondpad_cell_ref"
+            }
+          }
+          if {$expected_orient != [dict get $padcell bondpad orient]} {
+            utl::warn PAD 135 "Expected orientation ($expected_orient) of bondpad for padcell $padcell_name overridden with value [dict exists $padcell bondpad orient]"
+          }
+        }
+      } else {
+        if {[dict exists $library cells $bondpad_cell_ref orient]} {
+          if {[dict exists $library cells $bondpad_cell_ref orient $side_name]} {
+            dict set padcell bondpad orient [dict get $library cells $cell_ref orient $side_name]
+          } else {
+            if {[llength [dict get $library cells $bondpad_cell_ref orient]] == 1} {
+              dict set padcell bondpad orient [dict get $library cells $cell_ref orient]
+            } else {
+              utl::error PAD 136 "Unexpected value for orient attribute in library definition for $bondpad_cell_ref"
+            }
+          }
+        } else {
+          utl::error PAD 137 "No orientation information available for orientation of $cell_ref on side $side_name"
+        }
+      }
+      set bondpad_orient [dict get $padcell bondpad orient]
+      dict set padcell bondpad [verify_placement [dict get $padcell bondpad] $bondpad_cell_ref bondpad]
+    }
+    
+    # Verify bump 
+    if {[dict exists $padcell bump]} {
+      variable num_bumps_x
+      variable num_bumps_y
+
+      dict set padcell bump [check_bump [dict get $padcell bump]]
+
+      set row [dict get $padcell bump row]
+      set col [dict get $padcell bump col]
+    
+      if {$row < 1 || $row > $num_bumps_y} {
+        utl::error PAD 229 "The value for row is $row, but must be in the range 1 - $num_bumps_y"
+      }
+      if {$col < 1 || $col > $num_bumps_x} {
+        utl::error PAD 230 "The value for col is $col, but must be in the range 1 - $num_bumps_x"
+      }
+
+      set pitch [get_footprint_bump_pitch]
+      set origin [get_bump_origin $row $col]
```

Comment:
This is getting the origin of the bump, but for bumps that have origins in the LL corner this causes an issue where you cannot check the x/y's correctly because they are off by 1/2 bump width.

Reply:
Updated check to use get_bump_centre as the reference point